src/container.cpp

```diff
@@ -291,50 +289,50 @@ ReturnValue Container::queryAdd(int32_t index, const Thing& thing, uint32_t coun
 	}
 
 	if (!hasBitSet(FLAG_NOLIMIT, flags)) {
-		while (cylinder) {
-			if (cylinder == &thing) {
+		while (parent) {
+			if (parent == thing) {
 				return RETURNVALUE_THISISIMPOSSIBLE;
 			}
 
-			if (dynamic_cast<const Inbox*>(cylinder)) {
+			if (std::dynamic_pointer_cast<const Inbox>(parent)) {
 				return RETURNVALUE_CONTAINERNOTENOUGHROOM;
 			}
 
-			cylinder = cylinder->getParent();
+			parent = parent->getParent();
 		}
 
 		if (index == INDEX_WHEREEVER && size() >= capacity() && !hasPagination()) {
 			return RETURNVALUE_CONTAINERNOTENOUGHROOM;
 		}
 	} else {
-		while (cylinder) {
-			if (cylinder == &thing) {
+		while (parent) {
+			if (parent == thing) {
 				return RETURNVALUE_THISISIMPOSSIBLE;
 			}
 
-			cylinder = cylinder->getParent();
+			parent = parent->getParent();
 		}
 	}
 
-	const Cylinder* const topParent = getTopParent();
+	auto const topParent = getTopParent();
```

Comment:
Could this be `const auto const`
```suggestion
	const auto const topParent = getTopParent();
```